C/common/storage_asset_tracking.cpp

```diff
@@ -0,0 +1,274 @@
+		for (StorageAssetTrackingTuple* & rec : vec)
+		{
+			auto it = storageAssetTrackerTuplesCache.find(rec);
+			if (it == storageAssetTrackerTuplesCache.end())
+			{
+				// tuple not found in cache , so add it
+				storageAssetTrackerTuplesCache.insert(rec);
+			}
+			else
+			{
+				// tuple present and count value < count of reading, update cache
+				if ((*it)->m_maxCount < rec->m_maxCount)
+				{
+					storageAssetTrackerTuplesCache.erase(it);
+					storageAssetTrackerTuplesCache.insert(rec);
+				}
+				else if ((*it)->m_maxCount == rec->m_maxCount)
+				{
+					// case where counts are same but datapoints are different
+					// "a", "b", "c" and "a", "b", "foo"
+					// keep both the records
+					if ((*it)->m_datapoints.compare(rec->m_datapoints))
+					{
+						storageAssetTrackerTuplesCache.insert(rec);
+					}
+				}
+			}
+
+			Logger::getLogger()->debug("%s:%s Added storage asset tracker tuple to cache: '%s'", __FILE__, __FUNCTION__,
+					rec->assetToString().c_str());
+		}
+		delete (&vec);
+	}
+	catch (...)
+	{
+		Logger::getLogger()->error("%s:%s Failed to populate storage asset tracking tuples' cache", __FILE__, __FUNCTION__);
+		return;
+	}
+
+
+	return;
+}
+
+/**
+ * Find whether the Storage asset tracking tuple exists in the cache or not
+ *
+ * Return the pointer to the tuple 
+ *
+ * @param tuple        StorageAssetTrackingTuple Type
+ * @return	       A pointer to StorageAssetTrackingTuple in cache or null
+ */
+
+
+StorageAssetTrackingTuple* StorageAssetTracker::findStorageAssetTrackingCache(StorageAssetTrackingTuple& tuple)	
+{
+	StorageAssetTrackingTuple *ptr = &tuple;
+	std::unordered_multiset<StorageAssetTrackingTuple*>::const_iterator it = storageAssetTrackerTuplesCache.find(ptr);
+
+	if (it == storageAssetTrackerTuplesCache.end())
+	{
+	        Logger::getLogger()->debug("%s:%s :findStorageAssetTrackingCache tuple not found in cache ", __FILE__, __FUNCTION__);
+		return NULL;
+	}
+	else
+	{
+
+		   // tuple present and count value < count of reading, update cache
+                if ((*it)->m_maxCount < ptr->m_maxCount)
+                {
+			// record to be updated in tuple, delete old one 
+			Logger::getLogger()->debug("%s:%d:%s tuple present and count value < count of reading, update cache, erased dp%s ",  __FILE__,__LINE__, __FUNCTION__, (*it)->m_datapoints.c_str());
+
+	                storageAssetTrackerTuplesCache.erase(it);
+			return NULL;
+                }
+                else if ((*it)->m_maxCount == ptr->m_maxCount)
+                {
+                // case where counts are same but datapoints are different
+                // "a", "b", "c" and "a", "b", "foo"
+                // keep both the records
+        	        if (compareDatapoints(ptr->m_datapoints,(*it)->m_datapoints))
+               		{
+				//  record to be addded 
+				Logger::getLogger()->debug("%s:%d:%s tuple present and case where counts are same but datapoints are different, update cache ",  __FILE__,__LINE__, __FUNCTION__);
```

Comment:
Please remove __FILE__ from logs as complete path of source file would be stored into binary/library, which is not desirable.

Reply:
Fixed